### ISSUE TITLE
When setting off EMP explosions near a horde of pigs

### ISSUE THREAD

**Issue by Alicecomma** [NoImageAvailable/CombatExtended#570]
```
GrammarResolver failed to resolve a text (rootKeyword: r_logentry)
Custom rules:
  recipient_part0_label->rear right leg
  recipient_part_damaged0_label->rear right leg
  recipient_part1_label->rear left leg
  recipient_part_damaged1_label->rear left leg
  INITIATOR_label->81mm mortar shell (EMP)
  INITIATOR_definite->the 81mm mortar shell (EMP)
  INITIATOR_indefinite->a 81mm mortar shell (EMP)
  INITIATOR_possessive->its
  RECIPIENT_nameFull->a pig
  RECIPIENT_label->pig
  RECIPIENT_definite->the pig
  RECIPIENT_nameDef->the pig
  RECIPIENT_indefinite->a pig
  RECIPIENT_nameIndef->a pig
  RECIPIENT_pronoun->she
  RECIPIENT_possessive->her
  RECIPIENT_objective->her
  RECIPIENT_kind->sow

Includes:
  Combat_RangedDamage
  Combat_WoundIncludes

Constants:
  recipient_part0_destroyed: False
  recipient_part_damaged0_outside: True
  recipient_part1_destroyed: False
  recipient_part_damaged1_outside: True
  recipient_part_count: 2
  recipient_part_destroyed_count: 0
  recipient_part_damaged_count: 2
  deflected: False
  RECIPIENT_flesh: Normal
  ORIGINALTARGET_missing: True
  COVER_missing: True

0 ->[r_logentry] 
1    r_logentry->[RECIPIENT_definite]'s [destroyed_targets] was [destroyed_past] by [INITIATOR_definite]'s [WEAPON_projectile_label]. 
2       RECIPIENT_definite->the pig 
2       destroyed_targets->[recipient_part_destroyed0_label], [recipient_part_destroyed1_label], [recipient_part_destroyed2_label], and [recipient_part_destroyed3_label] UNRESOLVABLE: Cannot resolve subsymbol 'recipient_part_destroyed0_label'UNRESOLVABLE: Cannot resolve subsymbol 'recipient_part_destroyed1_label'UNRESOLVABLE: Cannot resolve subsymbol 'recipient_part_destroyed2_label'UNRESOLVABLE: Cannot resolve subsymbol 'recipient_part_destroyed3_label'UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       destroyed_past->punctured 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [destroyed_past] [RECIPIENT_definite]'s [destroyed_targets] [to] [destroyed_suffix]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       destroyed_past->pierced 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       to->into 
2       destroyed_suffix->bits 
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [missed] [ORIGINALTARGET_definite], [damaged_present] [RECIPIENT_definite]'s [damaged_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       missed->narrowly missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       damaged_present->injuring 
2       RECIPIENT_definite->the pig 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [destroyed_past] [RECIPIENT_definite]'s [destroyed_targets] and [damaged_past] [RECIPIENT_possessive] [damaged_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       destroyed_past->annihilated 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       damaged_past->pierced 
2       RECIPIENT_possessive->her 
2       damaged_targets->[recipient_part_damaged0_label], [recipient_part_damaged1_label], [recipient_part_damaged2_label], and [recipient_part_damaged3_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg UNRESOLVABLE: Cannot resolve subsymbol 'recipient_part_damaged2_label'UNRESOLVABLE: Cannot resolve subsymbol 'recipient_part_damaged3_label'
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
1    r_logentry->[RECIPIENT_definite]'s [damaged_targets] was [damaged_past] by [INITIATOR_definite]'s [WEAPON_projectile_label]. 
2       RECIPIENT_definite->the pig 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
2       damaged_past->shot 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [missed] [ORIGINALTARGET_definite], [destroyed_present] [RECIPIENT_definite]'s [destroyed_targets] and [damaged_present] [RECIPIENT_possessive] [damaged_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       missed->missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       destroyed_present->obliterating 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       damaged_present->shooting 
2       RECIPIENT_possessive->her 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
1    r_logentry->[RECIPIENT_definite]'s [destroyed_targets] was [destroyed_past] by [INITIATOR_definite]'s [expert] [WEAPON_projectile_label]. 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       destroyed_past->pierced 
2       INITIATOR_definite->the 81mm mortar shell (EMP) 
2       expert->apt UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [missed] [ORIGINALTARGET_definite], [destroyed_present] [RECIPIENT_definite]'s [destroyed_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       missed->missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       destroyed_present->obliterating 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [damaged_past] [RECIPIENT_definite]'s [damaged_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       damaged_past->pierced 
2       RECIPIENT_definite->the pig 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
1    r_logentry->[INITIATOR_definite] hit [RECIPIENT_definite] with a [WEAPON_projectile_label]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [damaged_past] [RECIPIENT_definite]'s [damaged_targets] [expertly]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       damaged_past->wounded 
2       RECIPIENT_definite->the pig 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
2       expertly->like a master 
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [missed] [ORIGINALTARGET_definite] and [destroyed_past] [RECIPIENT_definite]'s [destroyed_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       missed->narrowly missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       destroyed_past->annihilated 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] hit [RECIPIENT_definite]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       RECIPIENT_definite->the pig 
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [missed] [ORIGINALTARGET_definite] and [damaged_past] [RECIPIENT_definite]'s [damaged_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       missed->narrowly missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       damaged_past->wounded 
2       RECIPIENT_definite->the pig 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [destroyed_past] [RECIPIENT_definite]'s [destroyed_targets] [expertly]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       destroyed_past->punctured 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       expertly->with precision 
1    r_logentry->[INITIATOR_definite] hit [RECIPIENT_definite] with a [WEAPON_projectile_label] intended for [ORIGINALTARGET_definite]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
1    r_logentry->[RECIPIENT_definite]'s [damaged_targets] was [damaged_past] by [INITIATOR_definite]'s [expert] [WEAPON_projectile_label]. 
2       RECIPIENT_definite->the pig 
2       damaged_targets->[recipient_part_damaged0_label] and [recipient_part_damaged1_label] 
3          recipient_part_damaged0_label->rear right leg 
3          recipient_part_damaged1_label->rear left leg 
2       damaged_past->wounded 
2       INITIATOR_definite->the 81mm mortar shell (EMP) 
2       expert->adroit UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [missed] [ORIGINALTARGET_definite] and hit [RECIPIENT_definite]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       missed->narrowly missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       RECIPIENT_definite->the pig 
1    r_logentry->[INITIATOR_definite]'s [WEAPON_projectile_label] [destroyed_past] [RECIPIENT_definite]'s [destroyed_targets]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
2       destroyed_past->obliterated 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
1    r_logentry->[RECIPIENT_definite]'s [destroyed_targets] was [destroyed_past] into [destroyed_suffix] by [INITIATOR_definite]'s [WEAPON_projectile_label]. 
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'destroyed_targets'
2       destroyed_past->crushed 
2       destroyed_suffix->fragments 
2       INITIATOR_definite->the 81mm mortar shell (EMP) UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'
1    r_logentry->[INITIATOR_definite] [missed] [ORIGINALTARGET_definite] and hit [RECIPIENT_definite] with a [WEAPON_projectile_label]. 
2       INITIATOR_definite->the 81mm mortar shell (EMP) 
2       missed->missed UNRESOLVABLE: Cannot resolve subsymbol 'ORIGINALTARGET_definite'
2       RECIPIENT_definite->the pig UNRESOLVABLE: Cannot resolve subsymbol 'WEAPON_projectile_label'UNRESOLVABLE: Cannot resolve subsymbol 'r_logentry'
Verse.Log:Error(String, Boolean)
Verse.Log:ErrorOnce(String, Int32, Boolean)
Verse.Grammar.GrammarResolver:ResolveUnsafe(String, GrammarRequest, Boolean&, String, Boolean, Boolean)
Verse.Grammar.GrammarResolver:ResolveUnsafe(String, GrammarRequest, String, Boolean, Boolean)
Verse.Grammar.GrammarResolver:ResolveUnsafe(String, GrammarRequest, Boolean&, String, Boolean, Boolean)
Verse.Grammar.GrammarResolver:ResolveUnsafe(String, GrammarRequest, String, Boolean, Boolean)
Verse.Grammar.GrammarResolver:Resolve(String, GrammarRequest, String, Boolean, String)
Verse.LogEntry:ToGameStringFromPOV_Worker(Thing, Boolean)
Verse.LogEntry:ToGameStringFromPOV(Thing, Boolean)
Verse.DamageResult:AssociateWithLog(LogEntry_DamageResult)
CombatExtended.BulletCE:Impact(Thing)
CombatExtended.ProjectileCE:TryCollideWith(Thing)
CombatExtended.ProjectileCE:CheckCellForCollision(IntVec3)
CombatExtended.ProjectileCE:CheckForCollisionBetween()
CombatExtended.ProjectileCE:Tick()
Verse.TickList:Tick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()
```

**Comment by Alicecomma** [comment]
The issue here is that AmmoThing's explosion sets launcher=this. This means that the ammo thing is considered the pawn which shot the projectile, however everything in BattleLogEntry_RangedImpact attempts to find launcher.projectile.label, however of course launcher doesn't have a projectile tag.

Therefore, bullets shot from AmmoThing must receive their own separate BattleLogEntry which doesn't look for launcher.projectile.label.

**Comment by Alicecomma** [comment]
Fixed in #581 by adding a combat log handler for cookoff projectiles (which are defined by having a launcher ThingAmmo)
